modules/apps/collaboration/document-library/document-library-document-conversion/src/main/java/com/liferay/document/library/document/conversion/internal/upgrade/v1_0_0/UpgradeOpenOfficeConfiguration.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.document.library.document.conversion.internal.upgrade.v1_0_0;
+
+import com.liferay.document.library.document.conversion.configuration.OpenOfficeConfiguration;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.HashMapDictionary;
+import com.liferay.portal.kernel.util.PrefsProps;
+import com.liferay.portal.kernel.util.StringPool;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.Dictionary;
+
+import javax.portlet.PortletPreferences;
+
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+
+/**
+ * @author Pei-Jung Lan
+ */
+public class UpgradeOpenOfficeConfiguration extends UpgradeProcess {
+
+	public UpgradeOpenOfficeConfiguration(
+		ConfigurationAdmin configurationAdmin, PrefsProps prefsProps) {
+
+		_configurationAdmin = configurationAdmin;
+		_prefsProps = prefsProps;
+	}
+
+	protected void doUpgrade() throws Exception {
+		Configuration configuration = _configurationAdmin.getConfiguration(
+			OpenOfficeConfiguration.class.getName(), StringPool.QUESTION);
+
+		Dictionary properties = configuration.getProperties();
```

Comment:
It's possible we don't need to do this.  If this upgrade is triggering, the configuration won't have any property values, and they will all be overridden anyways by the code below.